packages/sales-api-service/src/services/exceptions/__tests__/exceptions.service.spec.js

```diff
@@ -110,4 +111,170 @@ describe('payment-journals service', () => {
       expect(result).toBeInstanceOf(PoclStagingException)
     })
   })
+
+  describe('createDataValidationError', () => {
+    let record, result
+    beforeEach(async () => {
+      record = {
+        id: 'test-id',
+        createTransactionPayload: {
+          dataSource: 'Post Office Sales',
+          serialNumber: '14345-48457J',
+          permissions: [{
+            licensee: {
+              firstName: 'Daniel',
+              lastName: 'Ricciardo',
+              organisation: 'Fishy Endeavours',
+              premises: '14 Howecroft Court',
+              street: 'Eastmead Lane',
+              town: 'Bristol',
+              postcode: 'BS9 1HJ',
+              country: 'GB',
+              birthDate: '1989-07-01',
+              email: 'daniel-ricc@example.com',
+              mobilePhone: '07722 123456',
+              preferredMethodOfNewsletter: 'Prefer not to be contacted',
+              preferredMethodOfConfirmation: 'Email',
+              preferredMethodOfReminder: 'Text'
+            },
+            permitId: 'test-permit-id',
+            startDate: '2021-06-15',
+            concessions: [{ type: 'Blue Badge', referenceNumber: '123456789' }]
+          }]
+        },
+        finaliseTransactionPayload: {
+          payment: {
+            timestamp: '2020-01-01T14:00:00Z',
+            amount: 30,
+            source: 'Post Office Sales',
+            channelId: '948594',
+            method: 'Cash'
+          }
+        },
+        stage: 'Staging',
+        createTransactionError: {
+          statusCode: 422,
+          error: 'Data validation error',
+          message: 'Error'
+        }
+      }
```

Comment:
Rather than declaring this here, please could you extract it into a function - getSampleRecord (or some similar name). Then in each test, it can get its own test data, rather than re-using.

Also, each test should do its own action, so 'maps the record to an instance of PoclDataValidationError' would be:
```
it('maps the record to an instance of PoclDataValidationError', () => {
  const record = getSampleRecord()
  const result = await createDataValidationError(record)
  expect(result).toBeInstanceOf(PoclDataValidationError)
})
```

Reply:
I have extracted it into a function. The record is created and then the `createDataValidationError` function is called in the `beforeEach` block, so would be called individually for each test

Reply:
Based on your comment below, I have changed this to a snapshot test